packages/aggregator/src/connections/EntityConnection.js

```diff
@@ -47,12 +63,12 @@ export class EntityConnection extends ApiConnection {
       query.ancestor_filter = `type:${aggregationEntityType}`;
     }
 
-    const response = await this.get(`hierarchy/${hierarchyName}/relationships`, query);
-
-    const formattedRelationships = {};
-    Object.entries(response).forEach(([descendant, ancestor]) => {
-      formattedRelationships[descendant] = { code: ancestor };
+    return this.runCachedFetch(`hierarchy/${hierarchyName}/relationships`, query).then(response => {
+      const formattedRelationships = {};
+      Object.entries(response).forEach(([descendant, ancestor]) => {
+        formattedRelationships[descendant] = { code: ancestor };
+      });
+      return [Object.keys(formattedRelationships), formattedRelationships];
```

Comment:
Am I being paranoid here that making this value return from a promise like this could lead to bugs where one caller mutates the object and then all other callers get affected by that? The return objects from this really really should be treated as immutable but there's no way of enforcing that.

Any good ideas on how to ensure that a new object gets returned any time a caller awaits this function?

Reply:
It's fair to be a bit paranoid, but this isn't an issue specific to this particular pattern, it's a js wide issue that's more obvious here. 

You can use https://immutable-js.github.io/immutable-js/ for immutable data structures, though I've never actually used it outside of some redux tutorials years ago. I'd say the unexpected interface might make it cause more issues than just keeping the POJO